core/src/main/java/org/apache/calcite/rex/RexBuilder.java

```diff
@@ -1917,6 +1917,34 @@ public RexLiteral makeZeroLiteral(RelDataType type) {
     return makeLiteral(zeroValue(type), type);
   }
 
+  public RexNode makeZeroForNestedType(RelDataType type) {
+    switch (type.getSqlTypeName()) {
+    case ARRAY:
```

Comment:
so you think a zero array is an array containing a zero?
or should it be an empty array?
the question is "zero for what algebraic operation"?
for concatenation the empty array is the zero.

Reply:
same question for multisets and maps

Reply:
I think an empty object makes more sense

Reply:
Yes, I also think what you said is very reasonable. Maybe I should implement an XxxZero for each nested type and implement the Comparable interface. I am trying to modify it. Could you give some suggestions?

Reply:
Why do you need the Comparable interface?
Just modify this function to return empty collections for ARRAY, MAP, MULTISET.

Reply:
Yes, but I think it will just work, no need to worry about it.
Just invoke the constructor for array with no arguments and cast it to the correct type.

Reply:
Got it, thanks a lot!

Reply:
Please take a look, I have addressed, thanks!